crates/bevy_render/src/mesh/primitives/dim2.rs

```diff
@@ -400,6 +400,55 @@ impl From<CircularSegment> for Mesh {
     }
 }
 
+/// A builder used for creating a [`Mesh`] with a [`ConvexPolygon`] shape.
+pub struct ConvexPolygonMeshBuilder<const N: usize> {
+    pub vertices: [Vec2; N]
+}
+
+impl<const N: usize> Meshable for ConvexPolygon<N> {
+    type Output = ConvexPolygonMeshBuilder<N>;
+
+    fn mesh(&self) -> Self::Output {
+        Self::Output {
+            vertices: self.vertices,
+        }
+    }
+}
+
+impl<const N: usize> MeshBuilder for ConvexPolygonMeshBuilder<N> {
+    fn build(&self) -> Mesh {
+        let mut indices = Vec::with_capacity(self.vertices.len() - 2);
+        let mut positions = Vec::with_capacity(self.vertices.len());
+
+        for vertex in self.vertices {
+            positions.push([vertex.x, vertex.y, 0.0]);
+        }
+        for i in 2..N as u32 {
+            indices.extend_from_slice(&[0, i - 1, i]);
+        }
```

Comment:
This is fine, though I wonder if placing an extra vertex at the centroid would be better since you'd probably get less long and thin triangles (rendering and collisions don't like them).

The same goes for our ellipses and circles as well. This isn't something we should handle here, but something to consider for future work

Reply:
Though you'd end up with more triangles per polygon. But it'd maybe still be worth it, I can't tell.